### PR TITLE
Rewrite stroke and fill shaders

### PR DESCRIPTION
**Pull request by eulertour**
## Changelog / Overview

Rewrite vectorized mobject shaders to be compatible with transformation matrices.

## Motivation

Adds `--use_projection_stroke_shaders` and `--use_projection_fill_shaders` flags which make vectorized mobjects rendered with OpenGL compatible with projection matrices. A second attempt at #1679 and #1680.

## Explanation for Changes

Shaders that can be transformed via matrices are easier to manipulate and the rewritten shaders are much easier to understand than the older ones.

[Documentation Reference](https://manimce--####.org.readthedocs.build)

## Testing Status

Tested with the following scene:
```py
class TextTest(Scene):
    def construct(self):
        import string
        text = OpenGLText(string.ascii_lowercase)
        text2 = OpenGLText(string.ascii_uppercase).next_to(text, DOWN)
        self.add(text, text2)
        self.interactive_embed()
```

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/renderer/opengl_renderer.py [hunk 1]
```diff
@@ -26,6 +27,12 @@
 )
 from .opengl_renderer_window import Window
 from .shader import Mesh, Shader
+from .vectorized_mobject_rendering import (
+    render_opengl_vectorized_mobject_fill,
+    render_opengl_vectorized_mobject_stroke,
+)
+
+count = 0
```
Comment (behackl) [inline manim/renderer/opengl_renderer.py:35]: Doesn't seem to be used.

**Review by behackl — COMMENTED** [review]
Neat, this time it works for me! 🥳 

image

I've had a brief look at the code (at least the non-glsl files), CLI-stuff etc. seems fine to me. I'm happy to have this merged as soon as some more people report that it works for them.

**Review by Nikhil0504 — COMMENTED** [review]
It looks fine to me now and the proportions look correct. But I did some more testing with the `use_projection_fill_shaders` and `use_projection_stroke_shaders` and found some glitches

Command Used:
`manim -p testing.py TextTest --renderer=opengl --use_projection_fill_shaders --write_to_movie `

Output:

https://user-images.githubusercontent.com/34582615/122899689-a9e02d00-d369-11eb-9da1-861329fdb85e.mp4

Command Used:
`manim -p testing.py TextTest --renderer=opengl --use_projection_stroke_shaders --write_to_movie `

Output:

https://user-images.githubusercontent.com/34582615/122899949-e4e26080-d369-11eb-8b08-239f2744dc65.mp4

Are we supposed to use both of them together or can we use them individually?

**Comment by Darylgolden** [comment]
> Are we supposed to use both of them together or can we use them individually?

I'm pretty sure they're meant to be used together (and eventually become the default shaders), so this isn't a problem.

File: manim/renderer/shaders/test/frag.glsl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#version 330
```
[the remark this answers is not in the document]
Comment (eulertour) [reply manim/renderer/shaders/test/frag.glsl:1]: This shader allows for easy testing of the OpenGL API without setting up projection matrices etc.

**Comment by eulertour** [comment]
@Darylgolden is right; this is expected because the rewritten shaders are projected differently than Grants.

**Review by Nikhil0504 — APPROVED** [review]
The shaders look good to me. Hopefully we use these new ones ASAP!

**Review by behackl — APPROVED** [review]
I'd like to ship the next release with this PR. Any objections, @eulertour?

**Comment by eulertour** [comment]
No, that'd be great

**Pull request merged**